core/contractsdk/go/example/award_manage/main.go

```diff
@@ -0,0 +1,253 @@
+package main
+
+import (
+	"errors"
+	"math/big"
+
+	"github.com/xuperchain/xuperchain/core/contractsdk/go/code"
+	"github.com/xuperchain/xuperchain/core/contractsdk/go/driver"
+	"github.com/xuperchain/xuperchain/core/contractsdk/go/utils"
+)
+
+const (
+	BALANCEPRE   = "balanceOf_"
+	ALLOWANCEPRE = "allowanceOf_"
+	MASTERPRE    = "admin"
+	TOTAL_SUPPLY = "TotalSupply"
+)
+
+type awardManage struct{}
+
+func (am *awardManage) Initialize(ctx code.Context) code.Response {
+	args := struct {
+		TotalSupply *big.Int `json:"totalSupply" gt:"0"`
+	}{}
+	if err := utils.Validate(ctx.Args(), &args); err != nil {
+		return code.Error(err)
+	}
+
+	caller := ctx.Initiator()
+	if len(caller) == 0 {
+		return code.Error(errors.New("missing caller"))
+	}
+	err := ctx.PutObject([]byte(BALANCEPRE+caller), []byte(args.TotalSupply.String()))
+	if err != nil {
+		return code.Error(err)
+	}
+
+	if err := ctx.PutObject([]byte(TOTAL_SUPPLY), []byte(args.TotalSupply.String())); err != nil {
+		return code.Error(err)
+	}
+
+	if err := ctx.PutObject([]byte(MASTERPRE), []byte(caller)); err != nil {
+		return code.Error(err)
+	}
+
+	return code.OK([]byte(args.TotalSupply.String()))
+}
+
+func (am *awardManage) AddAward(ctx code.Context) code.Response {
+	caller := ctx.Caller()
+	if caller == "" {
+		return code.Error(utils.ErrMissingCaller)
+	}
+	masterBytes, err := ctx.GetObject([]byte(MASTERPRE))
+	if err != nil {
+		return code.Error(err)
+	}
+	master := string(masterBytes)
+
+	if master != caller {
+		return code.Error(utils.ErrPermissionDenied)
+	}
+
+	args := struct {
+		Amount *big.Float `json:"amount" gt:"0"`
```

Comment:
Why use *big.Float here? Is *big.Int OK?

Reply:
Resolved!